src/python/pants/util/contextutil.py

```diff
@@ -403,3 +403,19 @@ def serve(port_queue, shutdown_queue):
   finally:
     shutdown_queue.put(True)
     t.join()
+
+
+@contextmanager
+def recursion_limit(limit):
+  """Temporarily modify the allowed recursion limit.
+
+  This method is intended to be used when serializing deeply nested structures such as dicts which
+  otherwise would raise a recursion error. It will not fix recursion errors due to cyclic
+  structures.
+  """
```

Comment:
A comment explaining what a normal `limit` is would be helpful context to know what the order of magnitutde is. On my system, Py2 and Py3 are both `1000`.

Reply:
Sure, but isn't the default value going to provide that information e.g. in `./pants options` and `./pants help`?

Reply:
It will provide that value to the user, but not to the Pants dev who may want to use this function somewhere you're not yet intending.

Reply:
Done!